data/scripts/sol-oic-gen.py

```diff
@@ -324,48 +334,60 @@ def get_field_enum_client_c(id, struct_name, name, prop):
     }
 
 def object_fields_from_repr_vec(name, props):
-    id = 0
     fields = []
-    for prop_name, prop in props.items():
+    type_to_fn = {
+        'string': get_field_string_client_c,
+        'integer': get_field_integer_client_c,
+        'number': get_field_number_client_c,
+        'boolean': get_field_boolean_client_c,
+    }
+
+    for id, (prop_name, prop) in enumerate(props.items()):
         if 'enum' in prop:
             fields.append(get_field_enum_client_c(id, name, prop_name, prop))
-        elif prop['type'] == 'string':
-            fields.append(get_field_string_client_c(id, prop_name, prop))
-        elif prop['type'] == 'integer':
-            fields.append(get_field_integer_client_c(id, prop_name, prop))
-        elif prop['type'] == 'number':
-            fields.append(get_field_number_client_c(id, prop_name, prop))
-        elif prop['type'] == 'boolean':
-            fields.append(get_field_boolean_client_c(id, prop_name, prop))
         else:
-            raise ValueError('unknown field type: %s' % prop['type'])
-        id += 1
+            fields.append(type_to_fn[prop['type']](id, prop_name, prop))
     return '\n'.join(fields)
 
 def generate_object_from_repr_vec_fn_common_c(name, props):
     fields_init = []
     for field_name, field_props in props.items():
         if 'enum' in field_props:
-            fields_init.append('.%s = state->%s,' % (field_name, field_name))
+            fields_init.append('        .%(name)s = state->%(name)s,' % {"name": field_name})
```

Comment:
No need to indent the generated code. If `indent` is installed, the whole file will be indented automatically so it can be manually inspected. This simplifies the code generator.